packages/storefront-lwc/src/modules/commerce/basket/basket.js

```diff
@@ -54,10 +54,14 @@ export default class Basket extends LightningElement {
     }
 
     removeHandler(event) {
-        const elem = event.srcElement;
-        ShoppingBasket.removeFromBasket(
-            parseInt(elem.getAttribute('data-item-index')),
-        );
-        this.products = ShoppingBasket.getCurrentBasket();
+        const itemId = event.srcElement.getAttribute('data-itemid');
+        ShoppingBasket.removeItemFromBasket(itemId)
+            .then(basket => {
+                this.basket = basket;
+                this.products = basket.products ? basket.products : [];
```

Comment:
Do we need to check to see if basket exists before calling `basket.products`?

Reply:
Remove can be called only if there are products in the basket. That means the basket has been created at this point. It will stay for as long as the session exists.